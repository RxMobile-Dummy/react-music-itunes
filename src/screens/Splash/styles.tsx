import { StyleSheet } from "react-native";
import { Colors } from "../../constants/Color";

const styles = StyleSheet.create({
  mainContainer: {
    backgroundColor: Colors.primary,
    flex: 1,
  },
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  muzikImg: {
    width: 300,
    height: 200,
  },
});
export default styles;
