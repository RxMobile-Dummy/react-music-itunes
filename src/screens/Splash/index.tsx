import { View, SafeAreaView, Image, Animated } from "react-native";
import React, { useEffect, useState } from "react";
import styles from "./styles";
import { Images } from "../../../assets/images";

interface Props {
  navigation?: any;
}

const Splash: React.FC<Props> = ({ navigation }) => {
  const [fadeAnim] = useState(new Animated.Value(0));

  useEffect(() => {
    _goToIntroScreen();
  }, []);

  // To animate the splash logo,
  useEffect(() => {
    Animated.timing(fadeAnim, {
      toValue: 1,
      duration: 4000,
      useNativeDriver: true,
    }).start();
  }, []);

  //Navigate to Into screen
  function _goToIntroScreen() {
    setTimeout(() => {
      //   navigation.navigate("Home");
      // }, 1000);
      navigation.navigate("Intro");
    }, 2000);
  }
  return (
    <SafeAreaView style={styles.mainContainer}>
      <View style={styles.container}>
        <Animated.View
          style={{
            opacity: fadeAnim,
          }}
        >
          <Image
            source={Images.SplashImg}
            style={[styles.muzikImg]}
            resizeMode={"contain"}
          />
        </Animated.View>
      </View>
    </SafeAreaView>
  );
};

export default Splash;
