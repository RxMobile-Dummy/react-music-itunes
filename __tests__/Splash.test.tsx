import React from "react";
import renderer from "react-test-renderer";
import Splash from "../src/screens/Splash";
import AsyncStorage from "@react-native-async-storage/async-storage";
jest.useFakeTimers();

const goBack = jest.fn();
const navigate = jest.fn();
const setOptions = jest.fn();

it("App renders without crashing", () => {
  const rendered = renderer.create(<Splash />).toJSON();
  expect(rendered).toBeTruthy();
});

it("App test against snapshot", () => {
  const tree = renderer.create(
    <Splash navigation={{ goBack, navigate, setOptions }} />
  );
  expect(tree).toMatchSnapshot();
});
it("checks if Async Storage is used", async () => {
  await asyncOperationOnAsyncStorage();

  expect(AsyncStorage.getItem).toBeCalledWith("myKey");
});
