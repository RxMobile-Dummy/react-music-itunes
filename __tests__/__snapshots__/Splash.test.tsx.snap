// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test against snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#0E0E0E",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "opacity": 0,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 200,
              "width": 300,
            },
          ]
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
